source/nodejs/adaptivecards-designer/src/designer-peers.ts

```diff
@@ -1384,14 +1384,24 @@ export abstract class DesignerPeer extends DraggableElement {
         return result;
     }
 
-    addElementsToDesignerSurface(designerSurface: HTMLElement, processChildren: boolean = false) {
-        designerSurface.appendChild(this.renderedElement);
-
+    addElementsToDesignerSurface(designerSurface: HTMLElement, processChildren: boolean = false, index: number = -1): number {
         if (processChildren) {
-            for (let i = 0; i < this.getChildCount(); i++) {
-                this.getChildAt(i).addElementsToDesignerSurface(designerSurface, processChildren);
+            let neighbor = designerSurface.children.item(index + 1);
+            designerSurface.insertBefore(this.renderedElement, neighbor);
```

Comment:
it's weird that js doesn't have a way to insert with index.

Reply:
Attempting to refactor to use https://developer.mozilla.org/en-US/docs/Web/API/Element/after so that we don't have to find the index